notebooker/serialization/mongo.py

```diff
@@ -91,9 +91,17 @@ def __init__(self, database_name="notebooker", mongo_host="localhost", result_co
         self.database_name = database_name
         self.mongo_host = mongo_host
         self.result_collection_name = result_collection_name
-        mongo_connection = self.get_mongo_database()
-        self.library = mongo_connection[result_collection_name]
-        self.result_data_store = gridfs.GridFS(mongo_connection, "notebook_data")
+
+        mongo_database = self.get_mongo_database()
+        self.library = mongo_database[result_collection_name]
+        self.result_data_store = gridfs.GridFS(mongo_database, "notebook_data")
+
+        conn = self.get_mongo_connection()
+        try:
+            conn.admin.command("enableSharding", self.database_name)
+            conn.admin.command({"shardCollection": f"{self.database_name}.notebook_data.chunks", "key": {"files_id": 1, "n": 1}})
+        except pymongo.errors.OperationFailure:
+            pass
```

Comment:
Worthy of a logger.error or logger.warning at least. I think we might instantiate this class quite often (i.e. almost every request that comes in) so maybe we should only attempt this once (at webapp startup in app.py perhaps)